supervisor/shared/external_flash/external_flash.c

```diff
@@ -57,13 +57,18 @@ static supervisor_allocation* supervisor_cache = NULL;
 
 // Wait until both the write enable and write in progress bits have cleared.
 static bool wait_for_flash_ready(void) {
-    uint8_t read_status_response[1] = {0x00};
     bool ok = true;
     // Both the write enable and write in progress bits should be low.
-    do {
-        ok = spi_flash_read_command(CMD_READ_STATUS, read_status_response, 1);
-    } while (ok && (read_status_response[0] & 0x3) != 0);
-    return ok;
+    #ifdef EXTERNAL_FLASH_QSPI_SINGLE
+        // For NVM without a ready bit in status register
+        return ok;
```

Comment:
Is this specific to the chip you're using? We could use `EXTERNAL_FLASH_QSPI_SINGLE` on a regular flash chip and this wouldn't be true, I think.

Could you add another bit to the features struct for flash chips that is something like `no_ready_bit` and check that?